src/features/todos.ts

```diff
@@ -1,8 +1,14 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction, Slice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
-export const todosSlice = createSlice({
+export const todosSlice: Slice<Todo[]> = createSlice({
   name: 'todos',
   initialState: [] as Todo[],
-  reducers: {},
+  reducers: {
+    setTodos(todos, action: PayloadAction<Todo[]>) {
+      todos.push(...action.payload);
```

Comment:
The `setTodos` reducer uses `push` to add todos, which appends the new todos to the existing state. If the intention is to replace the current list of todos, consider using `todos.splice(0, todos.length, ...action.payload)` or `return action.payload` to replace the state entirely.